### PR TITLE
fix(behavior_path_planner): fix getIntersectionTurnSignal func

### PR DESCRIPTION
**Pull request by k-obitsu**
Signed-off-by: k-obitsu <koichi.obitsu@tier4.jp>

## Description

Fix getIntersectionTurnSignal function of turn_signal_decider in the following points.
1. In case that the distance of 'turn_signal_distance' is shorter than the distance between two pass points.
Screenshot from 2022-03-22 20-32-17
2. in case that the distance of 'turn_signal_distance' is larger than the distance of 'intersection_search_distance'.

## Pre-review checklist for the PR author

PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has the write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@k-obitsu What kind of verification did you do to make sure this feature was working correctly?

**Comment by k-obitsu** [comment]
- The environment used for verification is as same as in the following JIRA ticket.
   https://tier4.atlassian.net/browse/EVT4-1753
    └VM(右折)：turnRIGHT_turn_signal_dist20m_base_idemitsu_anegasaki_r5.osm
    [https://eva.app.box.com/file/880875557814 - (https://eva.app.box.com/file/880875557814) 
    └VM(左折)：turnLEFT_turn_signal_dist20m_base_idemitsu_anegasaki_r5.osm 
    [https://eva.app.box.com/file/880873090253 - (https://eva.app.box.com/file/880873090253) 
    └PCD：inside.pcd - (https://eva.app.box.com/file/766597558549?s=4n2uyvws6jqevuxi89irow1d7q52296a)

- I have verified the following and confirmed that it works corectly.
    - Set the value of turn_signal_distance of the VM as follows.
        - turn signal distance = 20m : Turned on the turn signal at 15m   (intersection search distance is 15m) 
        - turn signal distance = 15m : Turned on the turn signal at 15m
        - turn signal distance = 5m : Turned on the turn signal at 5m
        - turn signal distance = 3m : Turned on the turn signal at 3m
        - turn signal distance = 1m : Turned on the turn signal at 1m
        - turn signal distance = 0m : Turned on the turn signal at 1m  (minimum distance of turn signal distance is 1m)

File: planning/behavior_path_planner/src/turn_signal_decider.cpp [hunk 1]
```diff
@@ -65,28 +65,48 @@ std::pair<TurnIndicatorsCommand, double> TurnSignalDecider::getIntersectionTurnS
...
+      }
+      if (turn_distance < turn_signal_distance_threshold_) {
+        turn_distance = turn_signal_distance_threshold_;
+      }
+      if (turn_distance < path_distance) {
+        judge_distance = turn_distance - path_distance;
+        turn_distance += turn_signal_distance_threshold_;
+      } else {
+        judge_distance = 0.0;
+      }
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/turn_signal_decider.cpp:100]: `judge_distance` should be defined in this scope, not outside this `for` loop.
Better to write like this
```
double judge_distance = 0.0;
if (turn_distance < path_distance) {
  judge_distance = turn_distance - path_distance;
  turn_distance += turn_signal_distance_threshold_;
}
```

File: planning/behavior_path_planner/src/turn_signal_decider.cpp [hunk 2]
```diff
@@ -65,28 +65,48 @@ std::pair<TurnIndicatorsCommand, double> TurnSignalDecider::getIntersectionTurnS
 
   // Get nearest intersection and decide turn signal
   double accumulated_distance = 0;
+  double turn_distance = intersection_search_distance_;
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/turn_signal_decider.cpp:68]: `turn_distance` should be defined in this scope, not outside this `for` loop.

File: planning/behavior_path_planner/src/turn_signal_decider.cpp [hunk 3]
```diff
@@ -65,28 +65,48 @@ std::pair<TurnIndicatorsCommand, double> TurnSignalDecider::getIntersectionTurnS
...
+  double path_distance = 0;
+  double judge_distance = 0;
 
   auto prev_point = path.points.front();
   auto prev_lane_id = lanelet::InvalId;
   for (const auto & path_point : path.points) {
-    accumulated_distance +=
-      tier4_autoware_utils::calcDistance3d(prev_point.point, path_point.point);
+    path_distance = tier4_autoware_utils::calcDistance3d(prev_point.point, path_point.point);
+    accumulated_distance += path_distance;
```
[the remark this answers is not in the document]
Comment (k-obitsu) [reply planning/behavior_path_planner/src/turn_signal_decider.cpp]: Fixed the above.

File: planning/behavior_path_planner/src/turn_signal_decider.cpp (hunk 1 above)
Comment (k-obitsu) [reply planning/behavior_path_planner/src/turn_signal_decider.cpp:100]: Fixed the above.

Comment (k-obitsu) [reply planning/behavior_path_planner/src/turn_signal_decider.cpp:100]: Fixed the above.

File: planning/behavior_path_planner/src/turn_signal_decider.cpp (hunk 2 above)
Comment (k-obitsu) [reply planning/behavior_path_planner/src/turn_signal_decider.cpp]: Fixed the above.

**Pull request closed without merging**